lib/geodb.js

```diff
@@ -4,61 +4,54 @@
 
 'use strict'
 
-var config = require('../config').get('geodb')
-var geodb = require('fxa-geodb')(config)
-var P = require('./promise')
-var ACCURACY_MAX_KM = 200
-var ACCURACY_MIN_KM = 25
+const config = require('../config').get('geodb')
+const geodb = require('fxa-geodb')(config)
+const ACCURACY_MAX_KM = 200
+const ACCURACY_MIN_KM = 25
 
 /**
 * Thin wrapper around geodb, to help log the accuracy
 * and catch errors. On success, returns an object with
 * `location` data. On failure, returns an empty object
 **/
-module.exports = function (log) {
-
+module.exports = log => {
   log.info({ op: 'geodb.start', enabled: config.enabled, dbPath: config.dbPath })
 
-  return function (ip) {
-    // this is a kill-switch and can be used to not return location data
+  return ip => {
     if (config.enabled === false) {
-      // if kill-switch is set, return a promise that resolves
-      // with an empty object
-      return new P.resolve({})
+      return {}
     }
-    return geodb(ip)
-      .then(function (location) {
-        var logEventPrefix = 'fxa.location.accuracy.'
-        var logEvent = 'no_accuracy_data'
-        var accuracy = location.accuracy
 
-        if (accuracy) {
-          if (accuracy > ACCURACY_MAX_KM) {
-            logEvent = 'unknown'
-          } else if (accuracy > ACCURACY_MIN_KM && accuracy <= ACCURACY_MAX_KM) {
-            logEvent = 'uncertain'
-          } else if (accuracy <= ACCURACY_MIN_KM) {
-            logEvent = 'confident'
-          }
-        }
+    try {
+      const location = geodb(ip)
+      const accuracy = location.accuracy
+      let confidence
+
+      if (accuracy > ACCURACY_MAX_KM) {
+        confidence = 'unknown'
+      } else if (accuracy > ACCURACY_MIN_KM && accuracy <= ACCURACY_MAX_KM) {
+        confidence = 'uncertain'
+      } else if (accuracy <= ACCURACY_MIN_KM) {
+        confidence = 'confident'
+      } else {
+        confidence = 'no_accuracy_data'
+      }
 
-        log.info({op: 'geodb.accuracy', 'accuracy': accuracy})
-        log.info({op: 'geodb.accuracy_confidence', 'accuracy_confidence': logEventPrefix + logEvent})
```

Comment:
We could probably update [graphs](https://kibana-fxa-us-west-2.prod.mozaws.net/#/dashboard/elasticsearch/FxA%20GeoDB%20Stats) to reflect new stats after shipping this.

Reply:
Oh, didn't realise there were graphs depending on this. Do you reckon I should put it back how it was? I thought one log line was better than two, but probably not if it breaks dashboard continuity.

Reply:
Fwiw I reverted this.